### DISCUSSION TITLE
Dynamic Segments

### DISCUSSION THREAD

**Discussion by wtcosta** [vercel/next.js#55250] [App Router]
I need to have several parameters in the url as in the example below, I would like to know how to handle this in routing without the need to create several folders

http://localhost/?id1=1&id2=2&id3=3&id4=4&id5=5&id6=6

**Comment by SidKH** [comment]
If you want to rely on query parameters you can access them with `searchParams`

```javascript
function Page({params, searchParams}) {
  console.log(searchParams.id1, searchParams.id2);
  return ...
}
```

Alternatively, check out [catch-all segments](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#catch-all-segments).
